src/components/spreadsheet/table-wrapper.tsx

```diff
@@ -494,8 +509,37 @@ const TableWrapper: FunctionComponent<TableWrapperProps> = ({
         if (gridRef.current?.api) {
             gridRef.current.api.setGridOption('rowData', equipments);
         }
-        setRowData(equipments);
-    }, [tabIndex, disabled, equipments]);
+
+        const equipmentType = toSpreadsheetEquipmentType(tablesNames[tabIndex]);
+
+        let equipmentsWithCustomColumnInfo = [...equipments];
+
+        Object.entries(additionalEquipmentsByNodesForCustomColumns).forEach((value) => {
+            const nodeAlias: string = value[0];
+            const equipmentsToAdd = value[1][equipmentType];
+            if (equipmentsToAdd) {
+                equipmentsToAdd.forEach((equipmentToAdd) => {
+                    let matchingEquipmentIndex = equipmentsWithCustomColumnInfo.findIndex(
+                        (equipmentWithCustomColumnInfo) => equipmentWithCustomColumnInfo.id === equipmentToAdd.id
+                    );
+                    let matchingEquipment = equipmentsWithCustomColumnInfo[matchingEquipmentIndex];
+                    if (matchingEquipment) {
+                        let equipmentWithAddedInfo: any = { ...matchingEquipment };
```

Comment:
I reckon it could be typed as follow :
`export interface RecursiveIdentifiable extends Identifiable {
    [alias: string]: Identifiable | unknown;
}
`